Added components.rb module; Updated in user_spec.rb, test scenario: V…

maksimok93 opened the pull request:
…erifies user can be deleted

maksimok93 committed:
Added components.rb module; Updated in user_spec.rb, test scenario: Verifies user can be deleted

maksimok93 committed:
Added test scenario to user_spec.rb: Verifies users role can be edited

maksimok93 committed:
Created export_page.rb, export_spec.rb; added test scenarios for checking if radio button is selected.

maksimok93 committed:
Added to export_spec.rb test scenario: Verifies if specific status in dropdown menu is selected

maksimok93 committed:
Added to user_spec.rb test scenarios to verify if buttons are enabled.

maksimok93 committed:
Added test scenarios to export_spec.rb, dashboard_spec.rb.

maksimok93 committed:
In user_spec.rb test data (first_name, last_name, website, password) has been moved to before(:each) block.

maksimok93 committed:
Added parameterized test with RADIO_BUTTONS array to export_spec.rb

maksimok93 committed:
Added messages_data; users_data text files. Added let() method to user_spec.rb

maksimok93 committed:
Project has been refactored according to rubocop warnings.

maksimok93 committed:
Added before class names top-level class documentation comments.

maksimok93 committed:
Added freeze method to RADIO_BUTTONS constant.

maksimok93 committed:
messages, users files has been moved to tests_data directory

AVzayats commented on spec/features/user_spec.rb:
```diff
@@ -43,9 +73,18 @@
 
+    it 'Verifies users role can be edited' do
+      @users_page.change_users_role('John', 'Editor')
+      expect(page).to have_content('Changed roles.')
```
it would also be good to check that real value is updated, since the message might be there, but value stays the same

AVzayats commented on spec/features/user_spec.rb:
```diff
@@ -43,9 +73,18 @@
+
     it 'Verifies user can be deleted' do
       @users_page.delete_user('John')
       expect(page).to have_content('User deleted.')
```
same as above, better to add verification that user is not displayed on the page

maksimok93 committed:
Style:

Move private method to the end of module.

maksimok93 committed:
Test:

Update test scenarios: Users role can de edited; User can be deleted. Add method to users_page.rb: get_users_role_xpath
